vyxal/helpers.py

```diff
@@ -294,6 +294,28 @@ def levenshtein_distance(s1: str, s2: str) -> int:
     return distances[-1]
 
 
+def local_minima(lhs: str) -> List[Union[int, float]]:
+    """Find the local minima of a mathematical function using Sympy"""
+
+    x = sympy.symbols("x")
+    d_dx = sympy.diff(sympy.sympify(lhs), x)
+    second_dx = sympy.diff(d_dx, x)
+    zeros = sympy.solve(d_dx, x)
+
+    return LazyList(z for z in zeros if second_dx.subs(x, z) > 0)
```

Comment:
There is also a local minimum if `f''(x) = 0` and `f'''(x) > 0`, right? And so on. [This](https://gist.github.com/tcibinan/8b7b58c6861a03752f8ec309e809d770) works.

Reply:
The way I learned it, you only need the first and second derivatives.

Reply:
@Lyxal I bet you this'll come back to bite as a bug at some point...